libs/core/algorithms/tests/unit/algorithms/uninitialized_relocate_backward.cpp

```diff
@@ -0,0 +1,308 @@
+//  Copyright (c) 2023 Isidoros Tsaousis-Seiras
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/init.hpp>
+#include <hpx/modules/testing.hpp>
+#include <hpx/parallel/algorithms/uninitialized_relocate.hpp>
+
+#define N 50
+#define M 20
+#define K 10
+
+using hpx::experimental::is_trivially_relocatable_v;
+using hpx::experimental::uninitialized_relocate_backward;
+
+struct trivially_relocatable_struct
+{
+    static int count;
+    static int move_count;
+    static int dtor_count;
+    int data;
+
+    explicit trivially_relocatable_struct(int data)
+      : data(data)
+    {
+        count++;
+    }
+    trivially_relocatable_struct(trivially_relocatable_struct&& other)
+      : data(other.data)
+    {
+        move_count++;
+        count++;
+    }
+    ~trivially_relocatable_struct()
+    {
+        dtor_count++;
+        count--;
+    }
+
+    // making sure the address is never directly accessed
+    friend void operator&(trivially_relocatable_struct) = delete;
+};
+int trivially_relocatable_struct::count = 0;
+int trivially_relocatable_struct::move_count = 0;
+int trivially_relocatable_struct::dtor_count = 0;
+
+HPX_DECLARE_TRIVIALLY_RELOCATABLE(trivially_relocatable_struct);
+static_assert(is_trivially_relocatable_v<trivially_relocatable_struct>);
+
+struct non_trivially_relocatable_struct
+{
+    static int count;
```

Comment:
For extra coverage: Instead of `count`, you _could_ store `static std::set<non_trivially_relocatable_struct*> objects_;`, `insert` in the ctor, `erase` in the dtor, and verify that you never construct over top of a live object nor destroy a dead object.